api/users/serializers.py

```diff
@@ -657,3 +669,104 @@ def update(self, instance, validated_data):
 
 class UserNodeSerializer(NodeSerializer):
     filterable_fields = NodeSerializer.filterable_fields | {'current_user_permissions'}
+
+
+class UserMessageSerializer(JSONAPISerializer):
+    """
+    Serializer for creating and managing `UserMessage` instances.
+
+    Attributes:
+        message_text (CharField): The text content of the message.
+        message_type (ChoiceField): The type of message being sent, restricted to `MessageTypes`.
+        institution (RelationshipField): The institution related to the message. Required.
+        user (RelationshipField): The recipient of the message.
+    """
+    id = IDField(read_only=True, source='_id')
+    message_text = ser.CharField(
+        required=True,
+        help_text='The content of the message to be sent.',
+    )
+    message_type = ser.ChoiceField(
+        choices=MessageTypes.choices,
+        required=True,
+        help_text='The type of message being sent. Must match one of the defined `MessageTypes`.',
+    )
+    institution = RelationshipField(
+        related_view='institutions:institution-detail',
+        related_view_kwargs={'institution_id': '<institution._id>'},
+        help_text='The institution associated with this message. This field is required.',
+    )
+    user = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<recipient._id>'},
+        help_text='The recipient of the message.',
+    )
+
+    def get_absolute_url(self, obj: UserMessage) -> str:
+        return api_v2_url(
+            'users:user-messages',
+            params={
+                'user_id': self.context['request'].parser_context['kwargs']['user_id'],
+                'version': self.context['request'].parser_context['kwargs']['version'],
+            },
+        )
+
+    def to_internal_value(self, data):
```

Comment:
this was caused by using the `RelationshipField` it doesn't handle POSTing to relationships well.